Mage/src/main/java/mage/abilities/common/BlocksCreatureTriggeredAbility.java

```diff
@@ -0,0 +1,56 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+
+/**
+ * @author Hiddevb
+ */
+public class BlocksCreatureTriggeredAbility extends TriggeredAbilityImpl {
+
+    private final FilterCreaturePermanent filter;
+
+    public BlocksCreatureTriggeredAbility(Effect effect, boolean optional) {
+        super(Zone.BATTLEFIELD, effect, optional);
+        this.filter = StaticFilters.FILTER_PERMANENT_CREATURE;
+    }
+
+    public BlocksCreatureTriggeredAbility(Effect effect, FilterCreaturePermanent filter, boolean optional) {
+        super(Zone.BATTLEFIELD, effect, optional);
+        this.filter = filter;
+    }
+
+    public BlocksCreatureTriggeredAbility(final BlocksCreatureTriggeredAbility ability) {
+        super(ability);
+        this.filter = ability.filter;
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.BLOCKER_DECLARED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return event.getSourceId().equals(this.getSourceId())
+                && !filter.match(game.getPermanent(event.getSourceId()), game);
```

Comment:
The filter should be matching against `event.getTargetId()`, not `event.getSourceId()`. The Target of a `BLOCKER_DECLARED` event is the creature being blocked, the Source is the creature blocking. Also, there shouldn't be an inversion operator--`filter.match` returns `true` if the object matches the filter.

Finally, invocations of `FilterPermanent.match` should use the version with four arguments (object, playerId, ability, game) whenever possible. The two-argument version can't evaluate some types of predicates and *silently ignores* them.

Reply:
Thanks for your review! I'm still figuring out how the code base exactly works, and I might have copied a bit too much code without thinking about it... I'll keep it in mind for future PR's

Reply:
Don't sweat it too much. There is a lot of "legacy code" in xmage that does things in deprecated or outright incorrect ways simply because it's a gigantic sprawling code base and there aren't nearly enough dev-hours to keep everything up to date.